src/accelerate/utils/modeling.py

```diff
@@ -233,6 +233,63 @@ def values(self):
         return sum([x[1:] for x in self], [])
 
 
+def check_tied_parameters(model: nn.Module):
+    """
+    Check if there is any indication in the given model that some weights should be tied.
+
+    Args:
+        model (nn.Module): The model to inspect
+
+    Returns:
+        bool: True if the model needs to have tied weights
+    """
+
+    # based on model.tie_weights() method
+    return any(
+        [
+            hasattr(model.config, True)
+            and getattr(model.config, "tie_word_embeddings", False)
+            and model.get_output_embeddings(),
+            hasattr(model.config, True)
+            and getattr(model.config, "is_encoder_decoder", False)
+            and getattr(model.config, "tie_encoder_decoder", False),
+            any(hasattr(module, "_tie_weights") for module in model.modules()),
+        ]
+    )
```

Comment:
Here maybe start with a check on whether one of the base classes if `PreTrainedModel` since this is very Transformers-centric. (`inspect.getmro(model)` will give you the list of super classes of the model)

Also no needs for `and` since you are using the `any`.